JOSESwift/Sources/CryptoImplementation/RSA.swift

```diff
@@ -50,24 +50,42 @@ fileprivate extension AsymmetricKeyAlgorithm {
         switch self {
         case .RSA1_5:
             return .rsaEncryptionPKCS1
+        case .RSAES_OAEP:
+            return .rsaEncryptionOAEPSHA1
         }
     }
 
     /// Checks if the plain text length does not exceed the maximum
     /// for the chosen algorithm and the corresponding public key.
     func isPlainTextLengthSatisfied(_ plainText: Data, for publicKey: SecKey) -> Bool {
+        let mLen: Int = plainText.count
+        let k = SecKeyGetBlockSize(publicKey)
         switch self {
         case .RSA1_5:
             // For detailed information about the allowed plain text length for RSAES-PKCS1-v1_5,
-            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2).
-            return plainText.count <= (SecKeyGetBlockSize(publicKey) - 11)
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2.1).
+            return mLen <= (k - 11)
+        case .RSAES_OAEP:
+            // For detailed information about the allowed plain text length for RSAES_OAEP,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.1.1).
+            // hLen = input limitation for the hash function (2^61 - 1 octets for SHA-1)
+            let hLen = (1 << 61) - 1
+            return mLen <= (k - 2 * hLen - 2)
         }
     }
 
     func isCipherTextLenghtSatisfied(_ cipherText: Data, for privateKey: SecKey) -> Bool {
         switch self {
         case .RSA1_5:
+            // For detailed information about the allowed cipher length for RSAES-PKCS1-v1_5,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.2.2).
             return cipherText.count == SecKeyGetBlockSize(privateKey)
+        case .RSAES_OAEP:
+            // For detailed information about the allowed cipher length for RSAES_OAEP,
+            // please refer to the RFC(https://tools.ietf.org/html/rfc3447#section-7.1.2).
+            // hLen = input limitation for the hash function (2^61 - 1 octets for SHA-1)
+            let hLen = (1 << 61) - 1
+            return cipherText.count == (2 * hLen + 2)
```

Comment:
That's a tricky one. I know the RFC-3447 [states](https://tools.ietf.org/html/rfc3447#section-7.1.2): 

>   C: ciphertext to be decrypted, an octet string of length k, where k = 2hLen + 2

However, there is an [errata](https://www.rfc-editor.org/errata_search.php?rfc=3447) that corrects that to

>  C: ciphertext to be decrypted, an octet string of length k, where k >= 2hLen + 2

[k denotes the length in octets of the RSA modulus n](https://tools.ietf.org/html/rfc3447#section-7.1.2), which you can get by calling [`SecKeyGetBlockSize`](https://developer.apple.com/documentation/security/1394222-seckeygetblocksize). So simply a check if `cipherText.cout == SecKeyGetBlockSize(privateKey)` should be enough here (assuming that the key has the correct length and is k >= 2hLen + 2).

Let me know what you think or if that doesn't make any sense. 🙂 

If I change the length checks accordingly, I can already successfully encrypt and decrypt a JWE using your `RSA-OAEP` implementation. 🎉

Reply:
Guys, just a quick update. The company I work for decided to go with CommonCrypto only so Im having a hard time here implementing this stuff. I'll still use the next weekend to finish this MR.

Reply:
Fixed